### DISCUSSION TITLE
Plans post Servlet TCK

### DISCUSSION THREAD

**Discussion by mnriem** [piranhacloud/piranha#2090] [Roadmap]
As we are inching ever closer to passing the Servlet TCK it is time to start thinking about what we do next?

Any thoughts?

**Comment by arjantijms** [comment]
Get some basics in place, like a Jakarta concurrency thread pool / executor service, and Jakarta transactions facility?

**Comment by mnriem** [comment]
What is the next TCK we should target?

**Comment by mnriem** [comment]
@Thihup Any comments?

**Comment by Thihup** [comment]
- Maybe the Jakarta Pages TCK
- Implement a way to connect to a database. I think it is a little limiting not to have a ConnectionPool/InitialContext and so on easily available to connect to it. As far as I know a lot of applications that are deployed in Tomcat/Jetty/Glassfish will need to connect to a datasource, and currently it is only possible if the user create a WebExtension for it.

**Reply by mnriem** [reply]
Are you aware of Manorrock Guppy? While it does not create the datasource it does make it easy to externalize connectivity to the database. See https://github.com/manorrock/guppy
